Index: core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/Assets.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/Assets.java b/core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/Assets.java
--- a/core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/Assets.java	(revision 73fe64cd4ae45638866299ef38be7af012147d57)
+++ b/core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/Assets.java	(revision 91ffc75d82834b417994b872e32d2d2c652fbac4)
@@ -297,6 +297,5 @@
 		public static final String SPIRIT_HAWK= "sprites/spirit_hawk.png";
 		public static final String RED_SENTRY= "sprites/red_sentry.png";
 		public static final String LIGHT_ORB= "sprites/light_orb.png";
-		public static final String LIGHT_ORB2= "sprites/light_orb.png";
 	}
 }
Index: core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/sprites/LightOrbSprite2.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/sprites/LightOrbSprite2.java b/core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/sprites/LightOrbSprite2.java
--- a/core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/sprites/LightOrbSprite2.java	(revision 73fe64cd4ae45638866299ef38be7af012147d57)
+++ b/core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/sprites/LightOrbSprite2.java	(revision 91ffc75d82834b417994b872e32d2d2c652fbac4)
@@ -2,7 +2,6 @@
 
 import com.shatteredpixel.shatteredpixeldungeon.Assets;
 import com.shatteredpixel.shatteredpixeldungeon.actors.mobs.npcs.LightOrb;
-import com.shatteredpixel.shatteredpixeldungeon.actors.mobs.npcs.LightOrb2;
 import com.shatteredpixel.shatteredpixeldungeon.effects.MagicMissile;
 import com.shatteredpixel.shatteredpixeldungeon.effects.Speck;
 import com.shatteredpixel.shatteredpixeldungeon.effects.particles.ShaftParticle;
@@ -16,7 +15,7 @@
     public LightOrbSprite2() {
         super();
 
-        texture( Assets.Sprites.LIGHT_ORB2 );
+        texture( Assets.Sprites.LIGHT_ORB );
 
         TextureFilm frames = new TextureFilm( texture, 14, 15 );
 
@@ -64,7 +63,7 @@
                 new Callback() {
                     @Override
                     public void call() {
-                        ((LightOrb2)ch).onZapComplete();
+                        ((LightOrb)ch).onZapComplete();
                     }
                 } );
         Sample.INSTANCE.play( Assets.Sounds.ZAP );
