Index: core/src/main/assets/messages/actors/actors.properties
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>ISO-8859-1
===================================================================
diff --git a/core/src/main/assets/messages/actors/actors.properties b/core/src/main/assets/messages/actors/actors.properties
--- a/core/src/main/assets/messages/actors/actors.properties	(revision a732a55e1e85902e7078c411cbef2304c4a0c174)
+++ b/core/src/main/assets/messages/actors/actors.properties	(revision a0b5a27bd4ebde27d4f22b522abc61096f374b17)
@@ -1228,3 +1228,5 @@
 actors.char.def_verb=dodged
 actors.char.invulnerable=invulnerable
 
+actors.mobs.lightorb$wndorbsmanage.orb1=Equip wand
+actors.mobs.lightorb$wndorbsmanage.move=Move
\ No newline at end of file
Index: core/src/main/assets/messages/windows/windows.properties
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>ISO-8859-1
===================================================================
diff --git a/core/src/main/assets/messages/windows/windows.properties b/core/src/main/assets/messages/windows/windows.properties
--- a/core/src/main/assets/messages/windows/windows.properties	(revision a732a55e1e85902e7078c411cbef2304c4a0c174)
+++ b/core/src/main/assets/messages/windows/windows.properties	(revision a0b5a27bd4ebde27d4f22b522abc61096f374b17)
@@ -298,9 +298,8 @@
 windows.wndwandmaker.berry=Oh, I see you have the berry! I do hope the rotberry plant didn't trouble you too much. As I promised, you can choose one of my high quality wands.
 windows.wndwandmaker.farewell=Good luck in your quest, %s!
 
-windows.wndorbsmanage.orb1=Equip wand
-windows.wndorbsmanage.move=Move
-
 windows.wndorboutfit.title=Choose a wand
 windows.wndorboutfit.desc=O caelesti sphaerae, quod artificium hoc accipere opus tuum permanere subsignata in auxilio opus nefaria.
 
+windows.lightorb$wndorbsmanage.orb1=Equip wand
+windows.lightorb$wndorbsmanage.move=Move
\ No newline at end of file
Index: core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/actors/mobs/npcs/LightOrb.java
===================================================================
diff --git a/core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/actors/mobs/npcs/LightOrb.java b/core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/actors/mobs/npcs/LightOrb.java
--- a/core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/actors/mobs/npcs/LightOrb.java	(revision a732a55e1e85902e7078c411cbef2304c4a0c174)
+++ b/core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/actors/mobs/npcs/LightOrb.java	(revision a0b5a27bd4ebde27d4f22b522abc61096f374b17)
@@ -29,7 +29,6 @@
 import com.shatteredpixel.shatteredpixeldungeon.ui.Window;
 import com.shatteredpixel.shatteredpixeldungeon.utils.GLog;
 import com.shatteredpixel.shatteredpixeldungeon.windows.WndOrbOutfit;
-import com.shatteredpixel.shatteredpixeldungeon.windows.WndOrbsManage;
 import com.watabou.noosa.Game;
 import com.watabou.noosa.audio.Sample;
 import com.watabou.noosa.tweeners.AlphaTweener;
@@ -37,7 +36,7 @@
 import com.watabou.utils.PathFinder;
 import com.watabou.utils.Random;
 
-public class LightOrb extends DirectableAlly {
+public class LightOrb extends Mob {
 
         private static final float TIME_TO_ZAP	= 1f;
 
@@ -174,7 +173,7 @@
                 }
         }
 
-        public class WndOrbsManage extends Window {
+        public static class WndOrbsManage extends Window {
 
                 private static final int WIDTH      = 120;
                 private static final int BTN_HEIGHT = 20;
Index: core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/items/weapon/melee/MagesStaff.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/items/weapon/melee/MagesStaff.java b/core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/items/weapon/melee/MagesStaff.java
--- a/core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/items/weapon/melee/MagesStaff.java	(revision a732a55e1e85902e7078c411cbef2304c4a0c174)
+++ b/core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/items/weapon/melee/MagesStaff.java	(revision a0b5a27bd4ebde27d4f22b522abc61096f374b17)
@@ -32,6 +32,7 @@
 import com.shatteredpixel.shatteredpixeldungeon.actors.hero.HeroSubClass;
 import com.shatteredpixel.shatteredpixeldungeon.actors.hero.Talent;
 import com.shatteredpixel.shatteredpixeldungeon.actors.mobs.Mob;
+import com.shatteredpixel.shatteredpixeldungeon.actors.mobs.npcs.LightOrb;
 import com.shatteredpixel.shatteredpixeldungeon.effects.particles.ElmoParticle;
 import com.shatteredpixel.shatteredpixeldungeon.items.ArcaneResin;
 import com.shatteredpixel.shatteredpixeldungeon.items.Item;
@@ -55,7 +56,6 @@
 import com.shatteredpixel.shatteredpixeldungeon.windows.WndBag;
 import com.shatteredpixel.shatteredpixeldungeon.windows.WndOptions;
 import com.shatteredpixel.shatteredpixeldungeon.windows.WndUseItem;
-import com.shatteredpixel.shatteredpixeldungeon.windows.WndOrbsManage;
 import com.watabou.noosa.audio.Sample;
 import com.watabou.noosa.particles.Emitter;
 import com.watabou.noosa.particles.PixelParticle;
@@ -147,7 +147,7 @@
 			else                             wand.cursed = false;
 			wand.execute(hero, AC_ZAP);
 		} else if (action.equals(AC_ORBS)) {
-			GameScene.show(new WndOrbsManage());
+			GameScene.show(new LightOrb.WndOrbsManage());
 			return;
 		}
 	}
Index: core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/windows/WndOrbsManage.java
===================================================================
diff --git a/core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/windows/WndOrbsManage.java b/core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/windows/WndOrbsManage.java
deleted file mode 100644
--- a/core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/windows/WndOrbsManage.java	(revision a732a55e1e85902e7078c411cbef2304c4a0c174)
+++ /dev/null	(revision a732a55e1e85902e7078c411cbef2304c4a0c174)
@@ -1,67 +0,0 @@
-package com.shatteredpixel.shatteredpixeldungeon.windows;
-
-import com.shatteredpixel.shatteredpixeldungeon.actors.Char;
-import com.shatteredpixel.shatteredpixeldungeon.actors.hero.HeroAction;
-import com.shatteredpixel.shatteredpixeldungeon.items.weapon.melee.MagesStaff;
-import com.shatteredpixel.shatteredpixeldungeon.messages.Messages;
-import com.shatteredpixel.shatteredpixeldungeon.scenes.GameScene;
-import com.shatteredpixel.shatteredpixeldungeon.ui.RedButton;
-import com.shatteredpixel.shatteredpixeldungeon.ui.Window;
-import com.watabou.noosa.Game;
-import com.watabou.utils.Callback;
-import com.shatteredpixel.shatteredpixeldungeon.actors.Char;
-
-import com.shatteredpixel.shatteredpixeldungeon.sprites.CharSprite;
-
-
-
-
-public class WndOrbsManage extends Window {
-
-    private static final int WIDTH      = 120;
-    private static final int BTN_HEIGHT = 20;
-    private static final int GAP        = 2;
-
-    protected RedButton btnorb1;
-    protected RedButton btnorb2;
-
-    public int pos = 0;
-
-    public CharSprite sprite;
-
-    public WndOrbsManage() {
-
-        super();
-
-
-        RedButton btnorb1 = new RedButton( Messages.get(this, "orb1") ) {
-            @Override
-            protected void onClick() {
-                GameScene.show(new WndOrbOutfit());
-            }
-
-        };
-        RedButton move = new RedButton( Messages.get(this, "move") ) {
-            @Override
-            protected void onClick() {
-                Game.runOnRenderThread(new Callback() {
-                    @Override
-                    public void call() {
-
-
-                    }
-                });
-            }
-        };
-
-        btnorb1.setRect( 0, GAP - 2 , WIDTH, BTN_HEIGHT );
-        move.setRect( 0, (int)btnorb1.bottom() + GAP, WIDTH, BTN_HEIGHT );
-        add( btnorb1 );
-        add( move );
-
-        resize( WIDTH, 42 );
-    }
-
-}
-
-
