Index: core/src/main/assets/messages/windows/windows.properties
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>ISO-8859-1
===================================================================
diff --git a/core/src/main/assets/messages/windows/windows.properties b/core/src/main/assets/messages/windows/windows.properties
--- a/core/src/main/assets/messages/windows/windows.properties	(revision 41d3f1e7e575db246427063ce469c7aa8c14ddfc)
+++ b/core/src/main/assets/messages/windows/windows.properties	(revision a732a55e1e85902e7078c411cbef2304c4a0c174)
@@ -302,5 +302,5 @@
 windows.wndorbsmanage.move=Move
 
 windows.wndorboutfit.title=Choose a wand
-windows.wndorboutfit.desc=test
+windows.wndorboutfit.desc=O caelesti sphaerae, quod artificium hoc accipere opus tuum permanere subsignata in auxilio opus nefaria.
 
Index: core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/actors/mobs/npcs/LightOrb.java
===================================================================
diff --git a/core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/actors/mobs/npcs/LightOrb.java b/core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/actors/mobs/npcs/LightOrb.java
--- a/core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/actors/mobs/npcs/LightOrb.java	(revision 41d3f1e7e575db246427063ce469c7aa8c14ddfc)
+++ b/core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/actors/mobs/npcs/LightOrb.java	(revision a732a55e1e85902e7078c411cbef2304c4a0c174)
@@ -25,7 +25,10 @@
 import com.shatteredpixel.shatteredpixeldungeon.scenes.GameScene;
 import com.shatteredpixel.shatteredpixeldungeon.sprites.CharSprite;
 import com.shatteredpixel.shatteredpixeldungeon.sprites.LightOrbSprite;
+import com.shatteredpixel.shatteredpixeldungeon.ui.RedButton;
+import com.shatteredpixel.shatteredpixeldungeon.ui.Window;
 import com.shatteredpixel.shatteredpixeldungeon.utils.GLog;
+import com.shatteredpixel.shatteredpixeldungeon.windows.WndOrbOutfit;
 import com.shatteredpixel.shatteredpixeldungeon.windows.WndOrbsManage;
 import com.watabou.noosa.Game;
 import com.watabou.noosa.audio.Sample;
@@ -170,5 +173,50 @@
                         return null;
                 }
         }
+
+        public class WndOrbsManage extends Window {
+
+                private static final int WIDTH      = 120;
+                private static final int BTN_HEIGHT = 20;
+                private static final int GAP        = 2;
+
+                protected RedButton btnorb1;
+                protected RedButton btnorb2;
+
+                public int pos = 0;
+
+                public CharSprite sprite;
+
+                public WndOrbsManage() {
+
+                        super();
+
+
+                        RedButton btnorb1 = new RedButton( Messages.get(this, "orb1") ) {
+                                @Override
+                                protected void onClick() {
+                                        GameScene.show(new WndOrbOutfit());
+                                }
+
+                        };
+                        RedButton move = new RedButton( Messages.get(this, "move") ) {
+                                @Override
+                                protected void onClick() {
+
+                                }
+                        };
+
+                        btnorb1.setRect( 0, GAP - 2 , WIDTH, BTN_HEIGHT );
+                        move.setRect( 0, (int)btnorb1.bottom() + GAP, WIDTH, BTN_HEIGHT );
+                        add( btnorb1 );
+                        add( move );
+
+                        resize( WIDTH, 42 );
+                }
+
+        }
+
+
+
 }
 
Index: core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/windows/WndOrbOutfit.java
===================================================================
diff --git a/core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/windows/WndOrbOutfit.java b/core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/windows/WndOrbOutfit.java
--- a/core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/windows/WndOrbOutfit.java	(revision 41d3f1e7e575db246427063ce469c7aa8c14ddfc)
+++ b/core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/windows/WndOrbOutfit.java	(revision a732a55e1e85902e7078c411cbef2304c4a0c174)
@@ -47,16 +47,13 @@
             protected void onClick(){
 
             }
-            };
+        };
 
-        btnWand.setRect( 75, message.top() + message.height() , BTN_SIZE, BTN_SIZE );
+        btnWand.setRect( 45, message.bottom() + 5, BTN_SIZE, BTN_SIZE );
         add( btnWand );
         btnWand.item(new WndBag.Placeholder(ItemSpriteSheet.WAND_HOLDER));
-        resize(WIDTH, 90);
-        };
-
-
-
+        resize(WIDTH, 76);
+        }
     }
 
 
