Index: core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/windows/WndOrbOutfit.java
===================================================================
diff --git a/core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/windows/WndOrbOutfit.java b/core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/windows/WndOrbOutfit.java
--- a/core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/windows/WndOrbOutfit.java	(revision c2a608c60a51c5c5e5bea10af55e830cd932c52d)
+++ b/core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/windows/WndOrbOutfit.java	(revision 1abb29fbb7c3053ed77199c8d56ad4b79f8d4944)
@@ -2,6 +2,7 @@
 
 import com.shatteredpixel.shatteredpixeldungeon.messages.Messages;
 import com.shatteredpixel.shatteredpixeldungeon.sprites.CharSprite;
+import com.shatteredpixel.shatteredpixeldungeon.sprites.ItemSpriteSheet;
 import com.shatteredpixel.shatteredpixeldungeon.ui.RedButton;
 import com.shatteredpixel.shatteredpixeldungeon.ui.Window;
 import com.watabou.noosa.Game;
@@ -12,9 +13,9 @@
     private static final int WIDTH      = 120;
     private static final int BTN_HEIGHT = 20;
     private static final int GAP        = 2;
+    private static final int BTN_SIZE	= 32;
 
-    protected RedButton btnorb1;
-    protected RedButton btnorb2;
+    protected WndBlacksmith.ItemButton btnWand;
 
     public int pos = 0;
 
@@ -24,34 +25,22 @@
 
         super();
 
-
-        RedButton btnorb1 = new RedButton( Messages.get(this, "orb1") ) {
-            @Override
-            protected void onClick() {
-
-            }
-
-        };
-        RedButton move = new RedButton( Messages.get(this, "move") ) {
+        btnWand = new WndBlacksmith.ItemButton(){
             @Override
-            protected void onClick() {
-                Game.runOnRenderThread(new Callback() {
-                    @Override
-                    public void call() {
-
+            protected void onClick(){
 
-                    }
-                });
             }
+            };
+
+        btnWand.setRect( 75 - BTN_SIZE, GAP, BTN_SIZE, BTN_SIZE );
+        add( btnWand );
+        btnWand.item(new WndBag.Placeholder(ItemSpriteSheet.WAND_HOLDER));
+        resize(WIDTH, (int)(btnWand.bottom() + GAP));
         };
 
-        btnorb1.setRect( 0, GAP - 2 , WIDTH, BTN_HEIGHT );
-        move.setRect( 0, (int)btnorb1.bottom() + GAP, WIDTH, BTN_HEIGHT );
-        add( btnorb1 );
-        add( move );
 
-        resize( WIDTH, 42 );
+
     }
 
-}
+
 
Index: docs/test.md
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/docs/test.md b/docs/test.md
new file mode 100644
--- /dev/null	(revision 1abb29fbb7c3053ed77199c8d56ad4b79f8d4944)
+++ b/docs/test.md	(revision 1abb29fbb7c3053ed77199c8d56ad4b79f8d4944)
@@ -0,0 +1,4 @@
+        btnorb1.setRect( 0, GAP - 2 , WIDTH, BTN_HEIGHT );
+        move.setRect( 0, (int)btnorb1.bottom() + GAP, WIDTH, BTN_HEIGHT );
+        add( btnorb1 );
+        add( move );
\ No newline at end of file
